superset-frontend/src/components/Datasource/DatasourceModal.tsx

```diff
@@ -139,7 +139,7 @@ const DatasourceModal: FunctionComponent<DatasourceModalProps> = ({
         ),
         columns: currentDatasource?.columns?.map(
           (column: Record<string, unknown>) => ({
-            id: column.id,
+            id: typeof column.id !== 'string' ? column.id : undefined,
```

Comment:
Since we only accept Number or undefined, maybe it'd be more clear to compare to `'number'`? Like `typeof column.id === 'number' ? column.id : undefined`

Reply:
I think so, let me fix it.

Reply:
Fixed